textworld/generator/__init__.py

```diff
@@ -113,22 +119,44 @@ def make_world_with(rooms, rng=None):
     return world
 
 
-def make_quest(world, quest_length, rng=None, rules_per_depth=(), backward=False):
-    state = world
-    if hasattr(world, "state"):
-        state = world.state
+def make_quest(world: Union[World, State], options: Optional[GameOptions] = None):
+    state = getattr(world, "state", world)
 
-    rng = g_rng.next() if rng is None else rng
+    if options is None:
+        options = GameOptions()
+
+        # By default, exclude quests finishing with: go, examine, look and inventory.
+        exclude = ["go.*", "examine.*", "look.*", "inventory.*"]
+        options.chaining.rules_per_depth = [options.kb.rules.get_matching(".*", exclude=exclude)]
+        options.chaining.rng = options.rngs['quest']
+
+    chains = []
+    for _ in range(options.nb_parallel_quests):
```

Comment:
This is how we can do "parallel" quests. I use quotes here because we do start from the resulting state of the previous quest, i.e. subsequent quests indirectly depend on the changes made by previous quests.

Reply:
The main issue is that with one `breadth` variable it is difficult to control the number of parallel quests vs. subquests. But I guess we could have a separate `root_breadth` variable in the chainer. Do you think it is equivalent to calling `sample_quest` multiple times?

Reply:
I guess I'm not sure what the difference between a subquest and a parallel quest is.

Reply:
For me, parallel quests don't share the same goal (e.g. `take obj | take key, unlock chest with key, open chest`), subquests do (e.g. `take obj, take key, unlock chest with key, open chest, insert obj in chest`). Does that makes more sense?